[docs]  Fix broken infrastructure link on EAS Build limitations page

mbaker3 opened the pull request:
# Why

Under the "Fixed memory and CPU limits on build worker servers" section the linked text "Server infrastructure reference" references a missing page.

# How

Added a `.` to the start of the path to reference the current directory rather than the root of the docs.

# Test Plan

Not tested locally but the change is trivial.

To reproduce:
1. Navigate to https://docs.expo.dev/build-reference/limitations/
2. Click on "Server infrastructure reference" under the "Fixed memory and CPU limits on build worker servers" section
3. Observe whether the browser navigates to a valid page.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

mbaker3 committed:
limitations.mdx - Fix broken infrastructure link

amandeepmittal commented on docs/pages/build-reference/limitations.mdx:
```diff
@@ -12,7 +12,7 @@ EAS Build is designed to work for any React Native project. However, it is good
 <Collapsible summary="Fixed memory and CPU limits on build worker servers">
 
-The [Server infrastructure reference](/infrastructure) contains the most up-to-date information about the current specifications of the Android (Ubuntu) and iOS (macOS) build servers. You may find that the resources available are not sufficient to build your app if your build process requires more than 12GB RAM. In the future we will be adding more powerful configurations to increase memory limit ...
+The [Server infrastructure reference](./infrastructure) contains the most up-to-date information about the current specifications of the Android (Ubuntu) and iOS (macOS) build servers. You may find that the resources available are not sufficient to build your app if your build process requires more than 12GB RAM. In the future we will be adding more powerful configurations to increase memory limi ...
```
```suggestion
The [Server infrastructure reference](/build-reference/infrastructure/) contains the most up-to-date information about the current specifications of the Android (Ubuntu) and iOS (macOS) build servers. You may find that the resources available are not sufficient to build your app if your build process requires more than 12GB RAM. In the future, we will add more powerful configurations to increase memory limits and speed up build times.
```

A small suggestion is to include the section path to in the URL. Also, a small typo fix (which is unrelated to the changes in this PR).

mbaker3 replied:
Thanks for the heads up!
If absolute paths are the convention for links in the docs should we go ahead and fix the other link in this file too?
https://github.com/expo/expo/blob/e515643f4bdd1ed45ec1625055b5bfdc61b9b624/docs/pages/build-reference/limitations.mdx?plain=1#L25

amandeepmittal replied:
Yes 😄

amandeepmittal reviewed (changes requested):
Thanks for fixing this @mbaker3! 
Left a small suggestion.

mbaker3 committed:
Convert internal links to absolute paths

Per @amandeepmittal's suggestions
 - All internal links on the Limitations page are now absolute paths
 - Minor typos fixed

mbaker3 commented:
Thanks for the feedback @amandeepmittal. Changes addressed.

amandeepmittal reviewed (approved):
Thank you!
